File: backend/src/main/java/com/github/andrepenteado/venda/domain/enums/FormaPagamento.java
/*
 * Autor: André Penteado
 * Criado em: 02/07/2026 -03
 * Observação: arquivo criado com ajuda da IA.
 */
package com.github.andrepenteado.venda.domain.enums;

/**
 * Formas de pagamento aceitas no PDV. O nome da constante é o valor gravado no banco.
 */
public enum FormaPagamento {

    DINHEIRO("Dinheiro"),
    PIX("PIX"),
    CARTAO_DEBITO("Cartão de débito"),
    CARTAO_CREDITO("Cartão de crédito");

    private final String descricao;

    FormaPagamento(String descricao) {
        this.descricao = descricao;
    }

    /**
     * Retorna a descrição apresentável da forma de pagamento.
     *
     * @return descrição da forma de pagamento.
     */
    public String getDescricao() {
        return descricao;
    }

}

File: frontend/src/app/domain/enums/forma-pagamento.ts
export enum FormaPagamento {
  DINHEIRO = 'DINHEIRO',
  PIX = 'PIX',
  CARTAO_DEBITO = 'CARTAO_DEBITO',
  CARTAO_CREDITO = 'CARTAO_CREDITO'
}

export const FormaPagamentoLabels: Record<FormaPagamento, string> = {
  [FormaPagamento.DINHEIRO]: 'Dinheiro',
  [FormaPagamento.PIX]: 'PIX',
  [FormaPagamento.CARTAO_DEBITO]: 'Cartão de débito',
  [FormaPagamento.CARTAO_CREDITO]: 'Cartão de crédito'
};
